v2/pkg/cli/delete.go

```diff
@@ -0,0 +1,308 @@
+		}
+	}
+	if len(o.Opts.Global.DeleteDestination) > 1 && !strings.Contains(o.Opts.Global.DeleteDestination, dockerProtocol) {
+		return fmt.Errorf("--destination flag must have a docker:// protocol prefix")
+	}
+
+	// remove file protocol from delete source
+	delete_file := o.Opts.Global.DeleteYaml
+
+	_, err := os.Stat(delete_file)
+	if len(o.Opts.Global.DeleteYaml) > 0 && !o.Opts.Global.DeleteGenerate && os.IsNotExist(err) {
+		return fmt.Errorf("file not found %s", delete_file)
+	}
+	return nil
+}
+
+// CompleteDelete - cobra complete
+func (o *ExecutorSchema) CompleteDelete() error {
+
+	if o.Opts.Global.DeleteGenerate {
+		o.Log.Debug("delete imagesetconfig file %s ", o.Opts.Global.ConfigPath)
+		// read and validate the DeleteImageSetConfiguration
+		cfg, err := config.ReadConfig(o.Opts.Global.ConfigPath, v1alpha2.DeleteImageSetConfigurationKind)
+		if err != nil {
+			return err
+		}
+		converted := cfg.(v1alpha2.DeleteImageSetConfiguration)
+		o.Log.Trace("delete imagesetconfig : %v ", converted)
+		if converted.Kind != "DeleteImageSetConfiguration" {
+			return fmt.Errorf("using the delete functionality requires the 'DeleteImageSetConfiguration' kind set in the yaml file")
+		}
+		// we now coerce deleteimagesetconfig to imagesetconfig
+		// as we want to use the underlying logic (filtering etc)
+		isc := v1alpha2.ImageSetConfiguration{
+			ImageSetConfigurationSpec: v1alpha2.ImageSetConfigurationSpec{
+				Mirror: v1alpha2.Mirror{
+					Platform:         converted.Delete.Platform,
+					Operators:        converted.Delete.Operators,
+					AdditionalImages: converted.Delete.AdditionalImages,
+				},
+			},
+		}
+		o.Config = isc
+	}
+
+	// update all dependant modules
+	mc := mirror.NewMirrorCopy()
+	o.Manifest = manifest.New(o.Log)
+	o.Mirror = mirror.New(mc, nil)
+
+	// logic to check mode
+	// always good to check - but this should have been detected in validate
+	if strings.Contains(o.Opts.Global.DeleteSource, fileProtocol) {
+		wd := strings.Split(o.Opts.Global.DeleteSource, fileProtocol)
+		o.Opts.Global.WorkingDir = filepath.Join(wd[1], workingDir)
+	} else {
+		return fmt.Errorf("--source flag must have a file:// protocol prefix")
+	}
+
+	// setup logs level, and logsDir under workingDir
+	err := o.setupLogsLevelAndDir()
+	if err != nil {
+		return err
+	}
+	if o.isLocalStoragePortBound() {
+		return fmt.Errorf("%d is already bound and cannot be used", o.Opts.Global.Port)
+	}
+	o.LocalStorageFQDN = "localhost:" + strconv.Itoa(int(o.Opts.Global.Port))
+
+	// ensure mirror and batch worker use delete logic
+	o.Opts.Function = string(mirror.DeleteMode)
+	o.Log.Info("executing %s ", o.Opts.Function)
+
+	if o.Opts.Global.DeleteGenerate {
+		absPath, err := filepath.Abs(o.Opts.Global.WorkingDir + deleteYaml)
+		if err != nil {
+			o.Log.Error("absolute path %v", err)
+		}
+		o.Log.Info("generate flag set, files will be created in %s", absPath)
```

Comment:
Shouldn't this log be updated a little bit based on the deleteID used?

Reply:
Took the lazy way out and changed it to the directory